include/lbann/layers/io/input/input_layer.hpp

```diff
@@ -99,6 +100,19 @@ template<>
 inline void input_layer<partitioned_io_buffer, data_layout::DATA_PARALLEL, El::Device::GPU>::validate_data_layout() {}
 #endif // LBANN_HAS_GPU
 
-}
+#ifndef LBANN_INPUT_LAYER_INSTANTIATE
+extern template class input_layer<
+  partitioned_io_buffer, data_layout::DATA_PARALLEL, El::Device::CPU>;
```

Comment:
I believe that the only valid input layer is data-parallel on CPU, at least currently.

Reply:
It is true that this is the only valid input layer, but this is, unfortunately, a dynamic check rather than a static one. These symbols are required for successful linkage.